gtest_parallel.py

```diff
@@ -358,6 +364,42 @@ def dump_to_file_and_close(self):
 
 # Record of test runtimes. Has built-in locking.
 class TestTimes(object):
+  class LockedFile(object):
+    def __init__(self, filename, mode):
+      self._filename = filename
+      self._mode = mode
+      self._fo = None
+
+    def __enter__(self):
+      self._fo = open(self._filename, self._mode)
+
+      try:
+        if sys.platform == 'win32':
+          msvcrt.locking(self._fo.fileno(), msvcrt.LK_LOCK, 1)
```

Comment:
Can you put a comment here saying that we are locking a fixed location inside the file, to use as an exclusive lock on the entire file? Also, are we sure that `_fo` is always at position 0 here? Otherwise different `_fo` handles will try to lock different segments. If `self._mode` here is append, this is wrong.